clang/lib/Index/USRGeneration.cpp

```diff
@@ -12,6 +12,7 @@
 #include "clang/AST/DeclCXX.h"
 #include "clang/AST/DeclTemplate.h"
 #include "clang/AST/DeclVisitor.h"
+#include "clang/AST/ODRHash.h"
```

Comment:
Is the header necessary?

Reply:
`ODRHash` is used inside the newly added code. Btw, what do you mean by "necessary"? I think one should not rely on transitive `#include`s in general. (I'm just an active contributor of [the IWYU project](https://github.com/include-what-you-use/include-what-you-use/). ~~Trust me!~~)

Reply:
Yep, this one looks that it's needed, sorry.

The idea (for the other comments), is that we try to avoid changes in PRs that are not strictly related to the change being made, mostly to avoid conflicts.

If we wanted to follow a IWYU style and reorganize our headers, we would prefer separate PRs or NFC changes (although if we wanted to sort that kind of policy, we would have  to do an RFC - again, because that sort of refactor is likely to induce conflicts).

Reply:
Just to clarify: do you insist that `#include` set added in a PR should be minimal for compilability?

Reply:
We shouldn't be minimizing the #include set, but it is often a 'red flag' when someone adds one, so Corentin brought it up.  

IMO, I have no problem with adding the include here, and a IWYU RFC would be welcomed (and I am less concerned about the 'induce conflicts' as these are touched rarely).